src/ORM/Marshaller.php

```diff
@@ -585,25 +585,26 @@ public function merge(EntityInterface $entity, array $data, array $options = [])
 
             if (isset($propertyMap[$key])) {
                 $value = $propertyMap[$key]($value, $entity);
-
                 // Don't dirty scalar values and objects that didn't
                 // change. Arrays will always be marked as dirty because
                 // the original/updated list could contain references to the
                 // same objects, even though those objects may have changed internally.
                 if (
-                    (
-                        is_scalar($value)
-                        && $original === $value
-                    )
-                    || (
-                        $value === null
-                        && $original === $value
-                    )
-                    || (
-                        is_object($value)
-                        && !($value instanceof EntityInterface)
-                        && $original == $value
-                    )
+                    is_scalar($value)
+                    && $original === $value
+                ) {
+                    continue;
+                }
+                if (
+                    $value === null
+                    && $original === $value
+                ) {
+                    continue;
+                }
+                if (
+                    is_object($value)
+                    && !($value instanceof EntityInterface)
+                    && $original == $value
```

Comment:
This is probably the only place where there's a reasonable readability improvement.

Reply:
https://github.com/cakephp/cakephp/pull/17797/files#diff-a0632c31b68e91238cf1ebe123a4a11b3abc5c0fd7f6d0781891609a851cd4d7L55
https://github.com/cakephp/cakephp/pull/17797/files#diff-3daf27c5efc6d12787f801a3c83a33b29a25254c243c59e934913166a107a077L719

These are improved too in my case

Reply:
Please just keep this change or else close this PR.